src/components/TodoList.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+import TodoItem from './TodoItem';
+
+interface TodoListProps {
+  todos: Todo[];
+  loading: boolean;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+}
+
+const TodoList: React.FC<TodoListProps> = ({ todos, loading, setTodos }) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {loading ? (
+        <div>
+          <div className="loader"></div>
+          <div className="loading">Loading ToDo&apos;s. Please wait...</div>
```

Comment:
Do you need this text? Check it on the working example

Reply:
No need, in example thats just 'Loading...' Just my vision. Deleted.